svgnative/src/SVGDocumentImpl.h

```diff
@@ -240,6 +247,7 @@ class SVGDocumentImpl
     // hierarchy.
     std::stack<StrokeStyleImpl> mStrokeStyleStack;
     std::stack<FillStyleImpl> mFillStyleStack;
+    std::vector<Rect> mBounds;
```

Comment:
I wonder if we really need to collect all bounds or if it wouldn't be better to compute union/intersection on the fly.

A document might be huge with many elements, better to free as many Rects as possible while traversing the tree.

Reply:
I agree that this is more compute intensive. The only reason why I didn't do them on the fly is because I have a debugging tool built as an example of SVG Native Viewer that consumes another API function that returns all of these rectangles and it becomes very useful to see which rectangle is messing up the bounds and where it comes from. Otherwise, one has to manually debug in GDB to figure out which shape is doing what. Of course that's a testing only thing so I can switch to doing this on the fly, saving all of these does cause extra memory.

Reply:
Yeah, unless it is too much code, you may put the debugging logic behind a debug preprocessor flag.